src/app/core/message-list/message-list.component.ts

```diff
@@ -34,6 +35,79 @@ export class MessageListComponent implements OnInit {
     this.sendMessage();
   }
 
+    quickRoll(r) {
+        let rollStr = ""
+        let rollRE = /^(\/r(oll)? )?(([0-9]+)[dD]([0-9]+)|0)?((kh|kl)1?)?((\+|\-)[0-9]+)? ?(.*)?/;
+        let m = rollRE.exec(this.messageInput);
```

Comment:
Tbh I'd suggest deconstructing this so it's easier to play with, and also using non-capturing groups so it doesn't care about things that are filler -

```suggestion
        const rollRE = /^(?:\/r(?:oll)? )?(?:([0-9]+)[dD]([0-9]+)|0)?(?:(kh|kl)1?)?([\+\-]?[0-9]+)? ?(.*)?/;
        const m = rollRE.exec(this.messageInput);
        
        // return if no m
        
        const [ num, dice, keep, plus, extra ] = m;
```

Then don't touch `m` itself directly again :-)

Reply:
That would certainly clean things up. I’ll look into it. Thanks.